mcci_catena_provision_chirpstack.py

```diff
@@ -0,0 +1,1280 @@
+
+    if sResult:
+        sPrefix = sResult.group(1)
+
+        sWord = re.search(r'\$\{(.*)\}', sLine)
+        sName = sWord.group(1)
+
+        if not sName in oAppContext.dVariables:
+            oAppContext.error("Unknown macro {}".format(sName))
+            sValue = '{' + sName + '}'
+        else:
+            sValue = oAppContext.dVariables[sName]
+
+        sResult = sPrefix + sValue
+
+    oAppContext.verbose("Expansion of {0}: {1}"
+                                .format(sLine, sResult)
+                                )
+    return sResult
+
+
+def doscript(sFileName):
+    '''
+    Perform macro expansion on a line of text.
+    The file is opened and read line by line.
+    Blank lines are ignored. Any text after a '#' character is treated as a
+    comment and discarded. Variables of the form ${name} are expanded. Any
+    error causes the script to stop.
+
+    Args:
+        sFileName: script name
+
+    Returns: 
+        True for script success, False for failure
+        
+    '''
+        
+    oAppContext.debug("DoScript: {}".format(sFileName))
+
+    try:
+        with open(sFileName, 'r') as rFile:
+            rFile = rFile.readlines()
+    except EnvironmentError as e:
+        oAppContext.error("Can't open file: {}".format(e))
+        return False
+
+    if not rFile:
+        oAppContext.error("Empty file")
+        return False
+
+    for line in rFile:
+        line = re.sub('\n$', '', line)
+        line = re.sub(r'^\s*#.*$', '', line)
+
+        line = expand(line)
+
+        if (re.sub(r'^\s*$', '', line) != ''):
+            if (oAppContext.fEcho):
+                sys.stdout.write(line + '\n')
+
+            if (oAppContext.fWriteEnable):
+                sResult = writecommand(
+                                        (re.sub('\n$', '', line)) + '\n'
+                                        )
+                if not (type(sResult) is tuple and sResult[0] is None):
+                    continue
+                else:
+                    oAppContext.error(
+                                        "Line: {0}\n\
+                                        Error: \n{1}"
+                                        .format(line, sResult[1])
+                    )
+                    return False
+    return True
+
+
+def closeport(sPortName):
+    '''
+    Close serial port
+
+    Args:
+        sPortName: serial port name
+
+    Returns: 
+        True if closed or None otherwise
+        
+    '''
+
+    if comPort.is_open:
+        comPort.reset_input_buffer()
+        comPort.reset_output_buffer()
+        comPort.close()
+        oAppContext.debug('Port {} closed'.format(sPortName))
+        return True
+    else:
+        oAppContext.error('Port {} already closed'.format(sPortName))
+        return None
+
+##############################################################################
+#
+#   main 
+#
+##############################################################################
+
+if __name__ == '__main__':
```

Comment:
please make a separate main function:

```python
def main():
   # body of main
   ''' all the code from lines 1094 to 1280 goes here

# standard wrapper
if __name__ == '__main__':
   main()
```

That way the main function is not special. You'll probably need to add `global` declaration for things like `pName`, `pDir`, etc.

Reply:
As per the suggestion, created separate main function